docs/software/git-software-development.md

```diff
@@ -163,3 +151,137 @@ A release of a software is created from your local clone of a software project.
+"fixes".
+
+#### Explain why a change was made in the commit message
+
+It is generally important also to explain _why_ a change was made.
+
+If this is not covered by the succinct subject line of your commit, you should
+explain the rationale behind your change in the commit message body.
+(The commit message starts with the subject line, then is optionally followed
+by a blank line plus the message body.)
+
+You can also explain in the commit message body _how_ this commit accomplishes
+the stated purpose in the subject - and you may find yourself needing to do
+this if there are some tricky details in the changes.
+But even if it is perfectly clear from the code and your commit message _what_
+you are changing and _how_ you are going about it, it is not always clear
+_why_ the change is needed or desired - so it is important to explain your
+reasons, in order to make this clear to the reviewer.
+
+#### Reference any relevant tickets
+
+Code changes often are related to a JIRA ticket, for instance SOFTWARE-1234.
+
+By referencing the name of a ticket in your pull request, it provides a
+convenient way to look into the background context for the change; and later
+on down the road, it makes it easy to find which changes were made for a
+particular task, referenced by the ticket name.
+
+Ideally, you can include a ticket reference each of these three places:
+
+1.  Your branch name.
+    For example, name the branch in your fork of the GitHub repo for the
+    pull request `SOFTWARE-1234.fix-memory-leak`.
+1.  Your commit messages.
+    For example, the subject of your commit message might read,
+    `fix a memory leak (SOFTWARE-1234)`.
+    If you have trouble squeezing the ticket name into the subject line,
+    or if you have a number of related tickets that you want to reference,
+    it is also OK to mention them later in the commit message body.
+1.  The pull request title.
+    If your pull request is just a single commit, and you have the ticket
+    reference in the subject line of the commit message, GitHub will include
+    this in the pull request title automatically.
+    But if you have multiple commits, or you have only included the ticket
+    reference in the body of the commit message, or more generally if you
+    want to tweak the title of the pull request, you should in any case make
+    a point to include the ticket name in the title of the pull request.
+    (By convention, we include this at the end of the title, in parentheses.)
+
+#### Further reading
+
+There are a number of articles and guides for making good git commits and
+good pull requests - a simple search will turn up plenty of material for
+the interested reader.
+
+See online guides such as
+[this one](https://dev.to/ruanbrandao/how-to-make-good-git-commits-256k)
+for more details.
+
+#### Brownie points
+
+You will get brownie points from Carl, personally, if you strive to make your
+code (and other text files) fit within an 80-column terminal window.
```

Comment:
Cute. I'll allow it.

Reply:
Thanks Boss.  It wouldn't have been a legit Strong Opinions section without it ;)